rtt/transports/corba/RemotePorts.cpp

```diff
@@ -252,12 +242,23 @@ bool RemoteOutputPort::disconnect(PortInterface* port)
     RemoteInputPort *portI = dynamic_cast<RemoteInputPort *>(port);
 
     if(portI == NULL){
-        Logger::In in("RemoteOutputPort::disconnect(PortInterface& port)");
-        log(Error) << "Port: " << port->getName() << " could not be disconnected from: " << this->getName()
-                << "because it could not be casted to type: RemoteInputPort" << endlog();
+        //in this case should be an InputPort at least.
+        RTT::base::InputPortInterface *portIn = dynamic_cast<RTT::base::InputPortInterface *>(port);
+        //if still NULL give up:
+        if(portIn == NULL){
+            Logger::In in("RemoteOutputPort::disconnect(PortInterface& port)");
+            log(Error) << "Port: " << port->getName() << " could not be disconnected from: " << this->getName()
+                    << "because it could not be casted to an InputPort type!" << endlog();
+            return false;
+        }
+
+        if(portIn->disconnect(this) or cmanager.disconnect(portIn)){
+            return true;
+        }
```

Comment:
So here is the problem, neither the local port
`portIn->disconnect(this)`
seem to be capable of disconnecting itself from a remote port nor
`cmanager.disconnect(portIn))`
the connection manager of the remote port can disconnect from the local port.
Since the flow interface disconnect function requires the "other" port to have a CDataFlowInterface as well, it can not be applied on the local port as also. So at least to me there was no obvious way how to realise the disconnect then.
Any suggestions?